libs/shared/base-editor-components/src/query-output/query-output.component.ts

```diff
@@ -21,18 +21,28 @@ import { QueryErrorComponent } from './query-error/query-error.component';
   standalone: true,
   imports: [FormsModule, HighlightjsWrapperComponent, QueryErrorComponent, TranslateModule],
 })
-export class QueryOutputComponent<T extends TableRow> extends SubscriptionHandler {
+export class QueryOutputComponent<T extends TableRow> extends SubscriptionHandler implements OnInit {
   private readonly clipboardService = inject(ClipboardService);
   private readonly modalService = inject(BsModalService);
+  private readonly validationService = inject(ValidationService);
 
   @Input() docUrl!: string;
   @Input() editorService!: EditorService<T>;
   @Output() executeQuery = new EventEmitter<string>();
   selectedQuery: 'diff' | 'full' = 'diff';
   private modalRef!: BsModalRef;
-
+  protected validationPassed: boolean = true;
   private readonly changeDetectorRef = inject(ChangeDetectorRef);
 
+  ngOnInit() {
+    this.subscriptions.push(
+      this.validationService.validationPassed$.subscribe((validationPassed: boolean) => {
+        this.validationPassed = validationPassed;
+        this.changeDetectorRef.detectChanges();
+      }),
+    );
+  }
```

Comment:
this is not necessary, you could use the `AsyncPipe` in the HTML template, like

```html
[disabled]="!(validationService.validationPassed$ | async)"
```

Remember to add `AsyncPipe` in the `imports: []` of this component